ibis/sql/postgres/compiler.py

```diff
@@ -584,7 +584,10 @@ def _literal(t, expr):
 
 def _day_of_week_index(t, expr):
     sa_arg, = map(t.translate, expr.op().args)
-    return sa.cast(sa.func.to_char(sa_arg, 'D'), sa.INTEGER)
+    return sa.cast(
+        sa.cast(sa.extract('dow', sa_arg) + 6, sa.SMALLINT) % 7,
+        sa.SMALLINT
+    )
```

Comment:
I wonder which one is better, implement on ibis expression level via rewrites (like You did in case of sqlite) or handle it at compilation?

Reply:
We still have to `extract(DOW)` here so I don't think we can rewrite to anything.